strongbox-web-core/src/main/java/org/carlspring/strongbox/interceptors/MavenArtifactControllerInterceptor.java

```diff
@@ -0,0 +1,130 @@
+package org.carlspring.strongbox.interceptors;
+
+import org.carlspring.strongbox.artifact.MavenArtifact;
+import org.carlspring.strongbox.artifact.MavenArtifactUtils;
+import org.carlspring.strongbox.controllers.layout.maven.MavenArtifactController;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.io.RepositoryPathResolver;
+import org.carlspring.strongbox.services.ConfigurationManagementService;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerMapping;
+import static org.carlspring.strongbox.storage.metadata.MetadataHelper.MAVEN_METADATA_XML;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class MavenArtifactControllerInterceptor
+        extends BaseArtifactControllerInterceptor
```

Comment:
@fuss86 am I understand correctly that inheritance needed here just because of `BaseArtifactControllerInterceptor.getStorage()` method? 
If so then I suggest to use `StoragesConfigurationManager.getStorage` in both interceptors instead.

Also we probably better to rename `BaseArtifactControllerInterceptor` into something more certain like `RepositoryInterceptor` or maybe `RepositoryRequestInterceptor`. 

Also we probably shouldn't use `Controller` keyword in `MavenArtifactControllerInterceptor` as well, because it's possible to have it not just for `MavenArtifactController` class.

Reply:
ok, I've renamed `BaseArtifactControllerInterceptor` to `ArtifactRequestInterceptor` and `MavenArtifactControllerInterceptor` to `MavenArtifactRequestInterceptor`

> If so then I suggest to use StoragesConfigurationManager.getStorage in both interceptors instead.

done

Reply:
> Also we probably better to rename BaseArtifactControllerInterceptor into something more certain like RepositoryInterceptor or maybe RepositoryRequestInterceptor.

`RepositoryRequestInterceptor` sounds the best, thanks